validphys2/src/validphys/results.py

```diff
@@ -96,12 +102,20 @@ def sqrtcovmat(self):
 
 
 class ThPredictionsResult(NNPDFDataResult):
-    def __init__(self, dataobj, stats_class, label=None):
+    """Class holding theory prediction
+    For legacy purposes it still accepts libNNPDF datatypes, but prefers python-pure stuff
+    """
+    def __init__(self, dataobj, stats_class, label=None, central_value=None):
         self.stats_class = stats_class
         self.label = label
-        self._std_error = dataobj.get_error()
-        self._rawdata = dataobj.get_data()
-        super().__init__(dataobj)
+        # Ducktype the input into numpy arrays
+        try:
+            self._std_error = dataobj.std(axis=1).to_numpy()
```

Comment:
I don't believe this is correct: If the constructor is accepting  a stats_class then that should be used to compute statistics. As discussed, ideally we would get rid of these objects and use `StatsResult` instead.

Reply:
The stats class was not being used here at all. Also, the plans for the `StatsResult` would probably need changes to PDF so that should wait for the changes to the PDF.

Reply:
But that line is incorrect for Hessian sets, no?

Reply:
because of replica 0?

Reply:
ohh, I see what you mean now. 
Because they are not necessarily symmetric and on top of that they don't necessarily represent the 68cl version. I'll fix that with the Stats Class which I guess is bringing all the ifs in the libNNPDF side